src/tests/unit/Observer.spec.ts

```diff
@@ -46,5 +46,21 @@ describe('Observer', () => {
                 })
             })
         })
+
+        it('should alert callback when Observer presents an error', async () => {
```

Comment:
Is this testing your change in the KvStore? We may want to add a case that specifically tests the 404 case somewhere in this test [KvStore.spec.ts](https://github.com/creditkarma/consul-client/blob/master/src/tests/integration/KvStore.spec.ts#L10). Also I'm not very familiar with this code so I might be missing something.

Reply:
God call out. This was just a test scenario that I happened to notice was missing in the suite; not something directly related to this PR, so I just added it. There appeared to be an existing [integration test](https://github.com/creditkarma/consul-client/blob/master/src/tests/integration/KvStore.spec.ts#L162) that addresses retries on the watch function (though it caps at 2 retries), so I still considered that a valid test for this scenario even though we're now retrying indefinitely.

Reply:
Gotcha thanks for the background!